src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -182,31 +231,58 @@ public class PullRequestHandler : IHttpHandler
             string.Format("checkout {0}", branch)
         };
 
-        return RunGitCommands(gitCommands, executionResult);
+        return RunGitCommands(gitCommands, gitRepoPath, executionResult);
+    }
+
+    // Gets a list of branches 
+    // - that are available in the repo (remoteBranches == false)
+    // - that are available in defined remotes (remoteBranches == true)
+    private bool GetBranches(string gitRepoPath, bool remoteBranches, out string[] branches, StringBuilder executionResult)
+    {
+        bool success = false;
+        branches = null;
+
+        StringBuilder branchesString = new StringBuilder();
+
+        if (remoteBranches)
+        {
+            success = RunGitCommands(new string[] { "branch --remote --list" }, gitRepoPath, branchesString);
+        }
+        else
+        {
+            success = RunGitCommands(new string[] { "branch --list" }, gitRepoPath, branchesString);
+        }
+
+        executionResult.Append(branchesString);
+
+        if (success)
+        {
+            branches = branchesString.ToString().Split(new char[] { '\n', '\r', ' ', '*' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
So do I need to do the same from browser? When should I use origin/master vs. upstream/master? Trying to same some typing... :)